staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -213,6 +213,33 @@ func newAuthenticator(c *cache, config *api.ExecConfig, cluster *clientauthentic
 	return c.put(key, a), nil
 }
 
+func isInteractive(isTerminalFunc func(int) bool, config *api.ExecConfig) (bool, error) {
+	var shouldBeInteractive bool
+	switch config.InteractiveMode {
+	case api.NeverExecInteractiveMode:
+		shouldBeInteractive = false
+	case api.IfAvailableExecInteractiveMode:
+		shouldBeInteractive = isTerminalFunc(int(os.Stdin.Fd())) && !config.StdinUnavailable
+	case api.AlwaysExecInteractiveMode:
+		if !isTerminalFunc(int(os.Stdin.Fd())) {
+			return false, errors.New("standard input is not a terminal")
+		}
+		if config.StdinUnavailable {
+			suffix := ""
+			if len(config.StdinUnavailableMessage) > 0 {
+				// only print extra ": <message>" if the user actually specified a message
+				suffix = fmt.Sprintf(": %s", config.StdinUnavailableMessage)
+			}
+			return false, fmt.Errorf("standard input is unavailable%s", suffix)
+		}
+		shouldBeInteractive = true
+	default:
+		return false, fmt.Errorf("unknown ExecInteractiveMode: %q", config.InteractiveMode)
```

Comment:
nit: use the external API field name (`interactiveMode`)

Reply:
also, should this be caught earlier in validation in `staging/src/k8s.io/client-go/tools/clientcmd/validation.go#validateAuthInfo`?

Reply:
this should be caught in https://github.com/kubernetes/kubernetes/blob/47e248f6fa40cad48ebc30340b6f2be339d25c27/staging/src/k8s.io/client-go/tools/clientcmd/validation.go#L311-L319

unless people are building their own `ExecConfig's` in code? i dunno how much this happens/how much we care. i seem to remember seeing only a handful of grep.app results that showed people doing this

if we did remove this validation, i'm not sure what we would do in the `default` case - seems like we need to return some sort of error if we get into that state (even tho we may never get into it)

i updated the error message in b8e58d6137b

Reply:
oh, I'm fine keeping the check here as well, just hoped it would be caught way earlier in kubeconfig validation as well